lib/LaTeXML/Engine/Base_ParameterTypes.pool.ltxml

```diff
@@ -96,6 +96,44 @@ DefParameterType('XToken', sub { $_[0]->readXToken; });
 # Read a number
 DefParameterType('Number', sub { $_[0]->readNumber; });
 
+# Note that LaTeX defines several utilities, fairly heavily used,
+#   \setcounter, \setlength, along with \addtocounter,\addtolength
+# which basically just assign a register to a Number or Dimension.
+# These are typically defined in LaTeXML as {Number}, {Dimension}, or [Number],[Dimension]
+# since they are read as a normal arg, but then reparsed as Number or Dimension.
+# The calc package REDEFINES these to support a bit of arithmetic, or extra braces.
+# Thus MANY commands suddenly accept a much wider range of expressions!
+# Note also that calc actually accepts Glue expressions for Dimensions.
+# [We could have calc redefine the ParameterTypes, but the internal data has already
+# been incorporated into definition's Parameters objects; Thus the current hackery:]
+
+# Read a regular arg, but content should be Number,
+# or extended calc syntax if calc.sty is loaded.
+DefParameterType('{Number}', sub {
+    my ($gullet) = @_;
+    my $value = $gullet->readArg();
+    return $gullet->readingFromMouth($value, sub {
+      (LookupValue('calc.sty.ltxml_loaded')
+       ? readCalcExpression($gullet, 'Number')
```

Comment:
This use of `readCalcExpression` will be a namespace headache once it reaches Rust. 

The `readCalcExpression` sub is only available to execute (in the perl sesne) _after_ the calc package has loaded, but it is already used in the Engine, before any bindings load. Ouch. Luckily there is late binding, but it's not easy to port.

Instead, one can have a method defined in Gullet (so always available early) with some generic name describing its behavior, maybe `readArithExpression` or such? And again use it only when calc is loaded. You may want to do this here, or I may just use my comment as a porting guide a bit later.